posts/deployToEcsWithGithubAction.md

```diff
@@ -0,0 +1,177 @@
+      id: task-def
+      uses: aws-actions/amazon-ecs-render-task-definition@v1
+      with:
+        task-definition: task-definition.json
+        container-name: dcdojo-container
+        image: ${{ steps.build-image.outputs.image }}
+
+    - name: Deploy Amazon ECS task definition
+      uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+      with:
+        task-definition: ${{ steps.task-def.outputs.task-definition }}
+        service: dc-dojo-service-github
+        cluster: dcdojo-cluster
+        wait-for-service-stability: true
+
+```
+
+This pipeline  will checkout the code, build the JAR file, build and push the Docker image to [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/), and then  deploy a new task definition to Amazon ECS, on every push to the master branch.
+
+Before using this pipeline, you will need to complete all the necessary infrastructure configuration on AWS:
+- Create an ECS repository
+- Create an ECS task definition, an ECS cluster, and an ECS service. [ECS getting started](https://us-east-2.console.aws.amazon.com/ecs/home?region=us-east-2#/firstRun)
+- Store your ECS task definition as a Json file in your repository
+
+The task-definition used in that project:
+
+task-definition.json
+```js
+{
+  "family": "dcdojo",
+  "executionRoleArn": "arn:aws:iam::112233445566:role/ecsTaskExecutionRole",
+  "containerDefinitions": [
+      {
+          "name": "dcdojo-container",
+          "image": "112233445566.dkr.ecr.us-east-2.amazonaws.com/dcdojo-amine:4622390a2b34f72y5f389cb62128a305641942cd",
+          "cpu": 512,
+          "portMappings": [
+            {
+              "hostPort": 8888,
+              "protocol": "tcp",
+              "containerPort": 8888
+            }
+          ],
+          "memory": 1024,
+          "essential": true,
+          "logConfiguration": {
+            "logDriver": "awslogs",
+            "options": {
+              "awslogs-group": "/ecs/dcdojo",
+              "awslogs-region": "us-east-2",
+              "awslogs-stream-prefix": "ecs"
+            }
+          }
+      }
+  ],
+  "requiresCompatibilities": [
+    "FARGATE"
+  ],
+  "networkMode": "awsvpc",
+  "cpu": "1024",
+  "memory": "2048"
+}
+```
+
+and a Dockerfile:
+
+```js
+FROM openjdk:8-jdk-alpine
+RUN apk --no-cache add curl
+VOLUME /tmp
+EXPOSE 8888
+ADD target/dcdojo-0.0.1-SNAPSHOT.jar app.jar
+ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","app.jar"]
+```
+
+After running this pipeline, the results looks like:
+
+![02](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/deployToEcsWithGithubAction-02.png)
+
+
+
+# Conclusion
+GitHub is working actively to integrate all the necessary DevOps features in their platform like GitHub Package to store your artifacts, and security alerts that track security vulnerabilities that apply to your project dependencies (Sofware Composition Analysis).
```

Comment:
GitHub is actively working to integrate all necessary DevOps features in their platform, such as GitHub Package, which stores your artifacts and security alerts that track security vulnerabilities that apply to your project dependencies (Software Composition Analysis).